portfolio/src/main/webapp/script.js

```diff
@@ -14,15 +14,34 @@
 
 "use strict"
 
-// loads comments
+// is called when page loads
 document.addEventListener("DOMContentLoaded", () => {
-    fetch("/comments").then((response) => (response.json())).then((json) => {
+    // set max-comments variable and change event on it
+    const DEFAULT_COMMENTS_NUMBER = 3;
+    let maxComments = DEFAULT_COMMENTS_NUMBER;
+    const maxCommentsElement = document.getElementById("max-comments");
+    maxCommentsElement.addEventListener("change", (event) => {
+        maxComments = parseInt(event.target.value);
+        // if parseInt couldn't parse the value or the value is negative
+        if (isNaN(maxComments) || maxComments < 0) {
+            maxComments = DEFAULT_COMMENTS_NUMBER;
+        }
+        // reload comments
+        loadComments(maxComments);
+    });
+    // initially load comments
+    loadComments(maxComments);
+});
+
+function loadComments(maxComments){
+    fetch(`/comments?maxcomments=${maxComments}`).then((response) => (response.json())).then((json) => {
```

Comment:
This line might be too long, maybe split it in more lines and add a comment explaining what it does?